projects/pgai/pgai/vectorizer/worker.py

```diff
@@ -40,6 +41,24 @@ class ApiKeyNotFoundError(Exception):
     pass
 
 
+def warn_on_old_version(version: Version):
+    if version.pgai_lib_version is None:
+        if (
+            version.ext_version is not None
+            and semver.Version.parse(version.ext_version) < "0.10.0"
+        ):
+            logger.warning(
+                "The pgai extension is outdated and can be upgraded. See https://github.com/timescale/pgai/blob/main/docs/vectorizer/migrating-from-extension.md"
+            )
+    else:
+        if semver.Version.parse(__version__) > semver.Version.parse(
+            version.pgai_lib_version
+        ):
+            logger.warning(
+                f"The pgai installation in your database is outdated and can be upgraded. Installed version: {version.pgai_lib_version}, latest version: {__version__}."  # noqa
+            )
```

Comment:
Should this check be somewhere more central in the library? This would be a valuable check to run in the text-to-sql code too.